plugins/wrapper/wrapper.go

```diff
@@ -0,0 +1,73 @@
+package wrapper
+
+import (
+	"github.com/gin-gonic/gin"
+)
+
+// PluginInstance is a wrapper to a real underlying plugin instance.
+type PluginInstance interface {
```

Comment:
Then I think there would be no way to determine whether a plugin has implemented a feature trait without using an error? Since we have an API that only detects the existence of plugin feature traits but does not execute them.

Reply:
The PluginInstance could have an method `Capabilities` which returns a string array.

Reply:
I'm sorry but I don't see a benefit in this. 
I guess you proposed this in order to simplify the process to get rid of the rather complicated wrapping logic in `github.com/gotify/server/plugin/loader`, but some signatures in the plugin API require a dedicated transformation, such as the interface for `Messenger` requires a `interface { SetMessageHandler(h papiv1.MessageHandler) }` and papiv1.MessageHandler is `interface { SendMessage(m Message) error }`, which requires an intermediate struct for transforming this API to the wrapper API.
Plus, I really love to use the wonderful type system in go to ensure the availability of methods instead of relying on an error or `strings.Contains`.

Reply:
oke